refactor!: consolidate shape types (#1337)
penrose/penrose: packages/docs-site/docs/ref/style/shapes-overview.md

## Strict Typing on Shape Parameters

Penrose enforces strict types on shape parameters. For example, we require the `r` field of a `Circle` shape to be a numerical value (type `FloatV`). If one writes,

```
myShape = Circle {
   r : true
}
```

or (by assignment or overriding)

```
myShape = Circle {}
override myShape.r = true
```

Penrose will report an error as follows.

```
Shape property myShape.r expects type FloatV and does not accept type BoolV.
```

The expected types of each field of each shape can be found within each shape's specification.

### Implicit Casting

For convenience, types which are similar can be casted into each other, implicitly. If type `A` implicitly casts into type `B`, then a shape field that expects value of type `A` will also accept values of type `B`. Currently, we support the following:

- `ListV` and `VectorV` are implicitly casted into each other.
- `MatrixV`, `LListV`, and `PtListV` are implicitly casted into each other.
- `TupV` can be casted into `ListV` and `VectorV`, but _not_ vice versa.

> Unfortunately, conversion of types into strings (typed `StrV`) are not yet supported.

Reviewer: ```suggestion
Conversion of types into strings (typed `StrV`) is not currently supported.
```
